internal/sbi/processor/nsselection_network_slice_information.go

```diff
@@ -10,89 +10,53 @@
 package processor
 
 import (
-	"encoding/json"
 	"fmt"
 	"math/rand"
 	"net/http"
-	"net/url"
-	"strings"
 
 	"github.com/gin-gonic/gin"
 
 	"github.com/free5gc/nssf/internal/logger"
-	"github.com/free5gc/nssf/internal/plugin"
 	"github.com/free5gc/nssf/internal/util"
 	"github.com/free5gc/openapi"
 	"github.com/free5gc/openapi/models"
 )
 
-// Parse NSSelectionGet query parameter
-func parseQueryParameter(query url.Values) (plugin.NsselectionQueryParameter, error) {
-	var (
-		param plugin.NsselectionQueryParameter
-		err   error
-	)
+type NetworkSliceInformationGetQuery struct {
+	// nolint: lll
+	NfType models.NrfNfManagementNfType `form:"nf-type" binding:"required,oneof=NRF UDM AMF SMF AUSF NEF PCF SMSF NSSF UDR LMF GMLC 5G_EIR SEPP UPF N3IWF AF UDSF BSF CHF NWDAF PCSCF CBCF HSS UCMF SOR_AF SPAF MME SCSAS SCEF SCP NSSAAF ICSCF SCSCF DRA IMS_AS AANF 5G_DDNMF NSACF MFAF EASDF DCCF MB_SMF TSCTSF ADRF GBA_BSF CEF MB_UPF NSWOF PKMF MNPF SMS_GMSC SMS_IWMSC MBSF MBSTF PANF IP_SM_GW SMS_ROUTER"`
 
-	if query.Get("nf-type") != "" {
-		param.NfType = new(models.NfType)
-		*param.NfType = models.NfType(query.Get("nf-type"))
-	}
+	NfId string `form:"nf-id" binding:"required,uuid"`
 
-	param.NfId = query.Get("nf-id")
+	// nolint: lll
```

Comment:
should be `",omitempty"`?

Reply:
https://github.com/gin-gonic/gin/blob/master/binding/form_mapping.go

I don't see any `omitempty` tag in the Gin form parser. Seems like the parser will set the default value by default(🤪). If `SliceInfoRequestForRegistration` is not provided in the form data, it will be set as a `nil` (which is the default value of a pointer).

In fact, the json parser set the default value as well, if the field is empty. The `omitempty` tag only do the job when we are marshaling the struct. It will not present the field in the marshal string if the field is being set as a default value (`""`, `0`, `nil`, etc).

As a result, this is a struct for parsing the query in requests. Not the one we want to send in responses. There's no need to add `json` struct tag or `omitempty` in this case.

---

Here's a simple demo. Without `omitempty`, the response shows the `name` field.
<img width="1029" alt="Screenshot 2024-11-14 at 1 13 51 PM" src="https://github.com/user-attachments/assets/a298baeb-1d10-4b60-859a-f8e612261748">

With `omitempty`, the response will omit fields with default values.
<img width="1024" alt="Screenshot 2024-11-14 at 1 14 57 PM" src="https://github.com/user-attachments/assets/48eb32e4-cbad-4bb9-a11e-49f7313f4683">